src/OpenTelemetry/Trace/ActivitySourceAdapter.cs

```diff
@@ -174,5 +174,10 @@ private void RunGetRequestedDataOtherSampler(Activity activity)
                 }
             }
         }
+
+        private bool IsParentSpanIdEmpty(Activity activity)
+        {
+            return activity.ParentSpanId.ToHexString() == "0000000000000000";
```

Comment:
I think this is only needed due to a bug, would you please add a comment similar to the below so we can circle back and clean it up in the future?

https://github.com/open-telemetry/opentelemetry-dotnet/blob/263f4ccf142af4d7e71d7cd87d6679e40554a209/src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/ActivityExtensions.cs#L92-L101

Reply:
Added, it would definitely be a more user-friendly API after

Reply:
Looking at the issue, it looks like already solved in rc2. Did you test with the new package?

Reply:
We should be removing this check as we already updated to rc2 now

Reply:
So basically what we're trying to do is call [SetParentId()](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/master/src/OpenTelemetry.Contrib.Extensions.AWSXRay/AWSXRayIdGenerator.cs#L60) to a root activity to change its trace id as its the only available API for us to modify trace id, and put `default` as its parent span id. However, calling `SetParentId()` will cause root activity's parent span id  non-default even through we set it as `default`. This happens in this [line](https://github.com/dotnet/runtime/blob/master/src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs#L490) where the parent span id is originally set as `default` but converted to an 0 string eventually. We just tested this on .NET 5 RC2 and after calling `SetParentId()` the parent span id is still `0000000000000000`.